src/databrowser.js

```diff
@@ -62,10 +62,21 @@ define([
             window_.preventDoubleScrolling(pane.find(".fd-scrollable"));
             vellum.datasources.on("error", function(event) {
                 var container = vellum.data.core.databrowser.errorContainer;
-                if (container && event.xhr.responseText) {
+                if (container) {
+                    // HQ will send a 400 (bad request) if it captures an error like an app build problem.
+                    // Unexpected errors are likely to send an HTML-based response, so show a generic message for them.
+                    var error = gettext("Could not load app properties. ");
```

Comment:
minor: might want to add the extra trailing space outside of the `gettext` call. I can see that being easily missed by translators.
```suggestion
                    var error = gettext("Could not load app properties.") + " ";
```